tests/e2e/cucumber/step_definitions/app.files.ts

```diff
@@ -99,6 +99,52 @@ When(
   }
 )
 
+When(
+  /^"([^"]*)" creates a public link to the following resource(s)? via the (sidebar panel|quick action)$/,
+  async function (
+    this: World,
+    stepUser: string,
+    _: string,
+    actionType: string,
+    stepTable: DataTable
+  ) {
+    const actor = this.actorsEnvironment.getActor({ id: stepUser })
+    const { publicLink: publicLinkPage } = new FilesPage({ actor })
+
+    const shareInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { resource, name, role, dateOfExpiration, password } = stepRow
+      if (!acc[resource]) {
+        acc[resource] = { name: '', role: '', dateOfExpiration: '', password: '' }
+      }
+      acc[resource].name = name
+      acc[resource].role = role
+      acc[resource].dateOfExpiration = dateOfExpiration
+      acc[resource].password = password
```

Comment:
what happens if we want to reuse this step to create public with default settings (provide `resource` name only in the data table)?